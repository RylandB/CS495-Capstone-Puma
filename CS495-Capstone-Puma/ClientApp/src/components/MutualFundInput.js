import React from 'react'

let state = {
    assets: [],
    inputName: null,
    inputPrice: null,
    inputQuantity: null,
    inputIncomePaymentMonth: null,
    inputIncomePaymentDay: null,
    inputUseDailyFactor: null,
    inputAccrualMethodType: null,
    inputExchangeType: null,
    inputEarningsPerShareDiluted: null,
    inputFundFamilyId: null,
    inputFundCategoryId: null,
    inputFundNumber: null,
    inputFundStatusType: null,
    inputShortTermRedemptionFeePercent: null,
    inputShortTermHoldingPeriod: null
};

export class MutualFundInput extends React.Component{
    componentDidMount(){
        console.log("component did mount")
        M.AutoInit();
    }

    constructor(props) {
        super(props);
        this.state = state;
        this.addAsset = this.addAsset.bind(this);
        this.removeAsset = this.removeAsset.bind(this);
        this.handleInputChange = this.handleInputChange.bind(this);
    }

    componentWillUnmount() {
        state = this.state;
    }

    renderTableHeader() {
        if(this.state.assets.length>0) {
            let header = Object.keys(this.state.assets[0]);
            return header.map((key, index) => {
                return <th key={index}>{key.toUpperCase()}</th>
            })
        }
    }

    renderTableData() {
        return this.state.assets.map((asset, index) => {
            const { name, price, quantity, incomePaymentMonth, 
                
                incomePaymentDay, useDailyFactor, accrualMethodType, 
                
                exchangeType, earningsPerShareDiluted, fundFamilyId, 
                
                fundCategoryId, fundNumber, fundStatusType, 
                
                shortTermRedemptionFeePercent, shortTermHoldingPeriod} = asset;
            return (
                <tr key={name}>
                    <td>{name}</td>
                    <td>{price}</td>
                    <td>{quantity}</td>
                    <td>{incomePaymentMonth}</td>
                    <td>{incomePaymentDay}</td>
                    <td>{useDailyFactor}</td>
                    <td>{accrualMethodType}</td>
                    <td>{exchangeType}</td>
                    <td>{earningsPerShareDiluted}</td>
                    <td>{fundFamilyId}</td>
                    <td>{fundCategoryId}</td>
                    <td>{fundNumber}</td>
                    <td>{fundStatusType}</td>
                    <td>{shortTermRedemptionFeePercent}</td>
                    <td>{shortTermHoldingPeriod}</td>
                </tr>
            )
        })
    }

    handleInputChange(event){

        const target = event.target;
        const value = event.target.value;
        const name = target.name;

        this.setState({
            [name]: value
        });

    }

    async addAsset(event){
        event.preventDefault();
        const newAsset = {
            name: this.state.inputName,
            price: this.state.inputPrice,
            quantity: this.state.inputQuantity,
            incomePaymentMonth: this.state.inputIncomePaymentMonth,
            incomePaymentDay: this.state.inputIncomePaymentDay,
            useDailyFactor: this.state.inputUseDailyFactor,
            accrualMethodType: this.state.inputAccrualMethodType,
            exchangeType: this.state.inputExchangeType,
            earningsPerShareDiluted: this.state.inputEarningsPerShareDiluted,
            fundFamilyId: this.state.inputFundFamilyId,
            fundCategoryId: this.state.inputFundCategoryId,
            fundNumber: this.state.inputFundNumber,
            fundStatusType: this.state.inputFundStatusType,
            shortTermRedemptionFeePercent: this.state.inputShortTermRedemptionFeePercent,
            shortTermHoldingPeriod: this.state.inputShortTermHoldingPeriod,
        };
        await this.setState({assets: this.state.assets.concat(newAsset)});
        this.props.mutualFundCallback(this.state.assets);
    }

    async removeAsset(){
        await this.setState({asset: this.state.assets.pop()});
        this.props.mutualFundCallback(this.state.assets);
    }

    render(){
        return(
            <div>
                <h3 id='title'>Mutual Funds</h3>
                <table className='table' id='assets'>
                    <tbody>
                    <tr>{this.renderTableHeader()}</tr>
                    {this.renderTableData()}
                    </tbody>
                </table>
                <form onSubmit={this.addAsset}>
                    
                <div className="row">
                    <div className = "input-field col s6">    
                        <label>Name</label>
                        <input type="text" name="inputName" required onChange={this.handleInputChange} value={this.state.inputName}/>
                    </div>
                </div>
                
                <div className = "row">
                    <div className = "input-field col s6">
                        <label>Price</label>
                        <input type="number" name="inputPrice" required onChange={this.handleInputChange} value={this.state.inputPrice}/>
                    </div>
                </div>
                
                <div className = "row">
                    <div className = "input-field col s6">    
                        <label>Quantity</label>
                        <input type="number" name="inputQuantity" required onChange={this.handleInputChange} value={this.state.inputQuantity}/>
                    </div>
                </div>
                   
                <div className = "row">
                    <div className = "input-field col s6">    
                        <label>IncomePaymentMonth</label>
                        <input type="number" name="inputIncomePaymentMonth" required onChange={this.handleInputChange} value={this.state.inputIncomePaymentMonth}/>
                    </div>
                </div>
                    
                <div className = "row">
                    <div className = "input-field col s6">    
                        <label>IncomePaymentDay</label>
                        <input type="number" name="inputIncomePaymentDay" required onChange={this.handleInputChange} value={this.state.inputIncomePaymentDay}/>
                    </div>
                </div>
                    
                <div className="row">
                    <div className = "input-field col s6">
                        <select name="inputUseDailyFactor" required onChange={this.handleInputChange} value={this.state.inputUseDailyFactor}>
                            <option value="true">true</option>
                            <option value="false">false</option>
                            <option value="null" selected="selected"> </option>
                        </select>
                        <label>UseDailyFactor</label>
                    </div>
                </div>
                    
                <div className = "row">
                    <div className = "input-field col s6">    
                        <label>AccrualMethodType</label>
                        <input type="text" name="inputAccrualMethodType" required onChange={this.handleInputChange} value={this.state.inputAccrualMethodType}/>
                    </div>
                </div>
                    
                <div className = "row">
                    <div className = "input-field col s6">    
                        <label>ExchangeType</label>
                        <input type="text" name="inputExchangeType" required onChange={this.handleInputChange} value={this.state.inputExchangeType}/>
                    </div>
                </div>
                    
                <div>
                    <div>    
                        <label>EarningsPerShareDiluted</label>
                        <input type="number" name="inputEarningsPerShareDiluted" required onChange={this.handleInputChange} value={this.state.inputEarningsPerShareDiluted}/>
                    </div>
                </div>
                
                <div>
                    <div>    
                        <label>FundFamilyId</label>
                        <input type="number" name="inputFundFamilyId" required onChange={this.handleInputChange} value={this.state.inputFundFamilyId}/>
                    </div>
                </div>
                
                <div>
                    <div>    
                        <label>FundCategoryId</label>
                        <input type="number" name="inputFundCategoryId" required onChange={this.handleInputChange} value={this.state.inputFundCategoryId}/>
                    </div>
                </div>
                
                <div>
                    <div>    
                <label>FundNumber</label>
                <input type="number" name="inputFundNumber" required onChange={this.handleInputChange} value={this.state.inputFundNumber}/>
                    </div>
                </div>
                
                <label>FundStatusType</label>
                <input type="text" name="inputFundStatusType" required onChange={this.handleInputChange} value={this.state.inputFundStatusType}/>
                <br />
                <label>ShortTermRedemptionFeePercent</label>
                <input type="number" name="inputShortTermRedemptionFeePercent" required onChange={this.handleInputChange} value={this.state.inputShortTermRedemptionFeePercent}/>
                <br />
                <label>ShortTermHoldingPeriod</label>
                <input type="number" name="inputShortTermHoldingPeriod" required onChange={this.handleInputChange} value={this.state.inputShortTermHoldingPeriod}/>
                <br />
                    <input type="submit" value="Add Asset" className="waves-effect waves-light btn-small"/>
                </form>
                <a onClick={this.removeAsset} className="waves-effect waves-light btn-small">Remove Asset</a>
                <br/>
                <br/>
            </div>
        );
    }

}