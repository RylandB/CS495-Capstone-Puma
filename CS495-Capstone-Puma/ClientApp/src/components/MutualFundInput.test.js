import React from "react";
import {MutualFundInput} from "./MutualFundInput";


it("AccountHistoryInput Renders Correctly", () => {
    const wrapper = shallow(
        <MutualFundInput />
    );

    expect(wrapper).toMatchSnapshot();
    
});