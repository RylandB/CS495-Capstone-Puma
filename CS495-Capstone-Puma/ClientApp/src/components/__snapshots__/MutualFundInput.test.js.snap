// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountHistoryInput Renders Correctly 1`] = `
<div>
  <h3
    id="title"
  >
    Mutual Funds
  </h3>
  <table
    className="table"
    id="assets"
  >
    <tbody>
      <tr>
        <th
          key="0"
        >
          NAME
        </th>
        <th
          key="1"
        >
          PRICE
        </th>
        <th
          key="2"
        >
          QUANTITY
        </th>
        <th
          key="3"
        >
          INCOMEPAYMENTMONTH
        </th>
        <th
          key="4"
        >
          INCOMEPAYMENTDAY
        </th>
        <th
          key="5"
        >
          USEDAILYFACTOR
        </th>
        <th
          key="6"
        >
          ACCRUALMETHODTYPE
        </th>
        <th
          key="7"
        >
          EXCHANGETYPE
        </th>
        <th
          key="8"
        >
          EARNINGSPERSHAREDILUTED
        </th>
        <th
          key="9"
        >
          FUNDFAMILYID
        </th>
        <th
          key="10"
        >
          FUNDCATEGORYID
        </th>
        <th
          key="11"
        >
          FUNDNUMBER
        </th>
        <th
          key="12"
        >
          FUNDSTATUSTYPE
        </th>
        <th
          key="13"
        >
          SHORTTERMREDEMPTIONFEEPERCENT
        </th>
        <th
          key="14"
        >
          SHORTTERMHOLDINGPERIOD
        </th>
      </tr>
    </tbody>
  </table>
  <form
    onSubmit={[Function]}
  >
    <label>
      Name
    </label>
    <input
      name="inputName"
      onChange={[Function]}
      required={true}
      type="text"
      value={null}
    />
    <br />
    <label>
      Price
    </label>
    <input
      name="inputPrice"
      onChange={[Function]}
      required={true}
      type="number"
      value={null}
    />
    <br />
    <label>
      Quantity
    </label>
    <input
      name="inputQuantity"
      onChange={[Function]}
      required={true}
      type="number"
      value={null}
    />
    <br />
    <label>
      IncomePaymentMonth
    </label>
    <input
      name="inputIncomePaymentMonth"
      onChange={[Function]}
      required={true}
      type="number"
      value={null}
    />
    <br />
    <label>
      IncomePaymentDay
    </label>
    <input
      name="inputIncomePaymentDay"
      onChange={[Function]}
      required={true}
      type="number"
      value={null}
    />
    <br />
    <label>
      UseDailyFactor
    </label>
    <select
      name="inputUseDailyFactor"
      onChange={[Function]}
      required={true}
      value={null}
    >
      <option
        value="true"
      >
        true
      </option>
      <option
        value="false"
      >
        false
      </option>
      <option
        selected="selected"
        value="null"
      >
         
      </option>
    </select>
    <br />
    <label>
      AccrualMethodType
    </label>
    <input
      name="inputAccrualMethodType"
      onChange={[Function]}
      required={true}
      type="text"
      value={null}
    />
    <br />
    <label>
      ExchangeType
    </label>
    <input
      name="inputExchangeType"
      onChange={[Function]}
      required={true}
      type="text"
      value={null}
    />
    <br />
    <label>
      EarningsPerShareDiluted
    </label>
    <input
      name="inputEarningsPerShareDiluted"
      onChange={[Function]}
      required={true}
      type="number"
      value={null}
    />
    <br />
    <label>
      FundFamilyId
    </label>
    <input
      name="inputFundFamilyId"
      onChange={[Function]}
      required={true}
      type="number"
      value={null}
    />
    <br />
    <label>
      FundCategoryId
    </label>
    <input
      name="inputFundCategoryId"
      onChange={[Function]}
      required={true}
      type="number"
      value={null}
    />
    <br />
    <label>
      FundNumber
    </label>
    <input
      name="inputFundNumber"
      onChange={[Function]}
      required={true}
      type="number"
      value={null}
    />
    <br />
    <label>
      FundStatusType
    </label>
    <input
      name="inputFundStatusType"
      onChange={[Function]}
      required={true}
      type="text"
      value={null}
    />
    <br />
    <label>
      ShortTermRedemptionFeePercent
    </label>
    <input
      name="inputShortTermRedemptionFeePercent"
      onChange={[Function]}
      required={true}
      type="number"
      value={null}
    />
    <br />
    <label>
      ShortTermHoldingPeriod
    </label>
    <input
      name="inputShortTermHoldingPeriod"
      onChange={[Function]}
      required={true}
      type="number"
      value={null}
    />
    <br />
    <input
      type="submit"
      value="Add Asset"
    />
  </form>
  <button
    onClick={[Function]}
  >
     Remove Asset
  </button>
  <br />
  <br />
</div>
`;
